matrix-synapse: init at 0.12.0

roblabla opened the pull request:
Adds matrix.org's synapse server as a package and module.

I am currently using this package on my server without any problems.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/nixos/nixpkgs/12301)

<!-- Reviewable:end -->

roblabla committed:
matrix-synapse: init at 0.12.0

lethalman commented:
That's a huge addition, thanks!

Please check the travis error, daemonize doesn't build. Make sure you have enabled nix chroots. Also note that the uid 219 is no more available.

roblabla commented:
I changed the uid to 223, and fixed a build error concerning pydenticon missing mock (caught that by enabling chroot builds).

Now I'm having a small problem. The sha256 returned by fetchgit seems to have changed from under my feets, even though the tag still points to the same commit. Any idea how this could happen ? https://github.com/roblabla/nixpkgs/blob/feature-initMatrixSynapse/pkgs/servers/matrix-synapse/default.nix#L17-L21

Yesterday : 29f3d95c0a5a0c902a1c37357dea1a9150d3177d17f1045e27a9d01692816af9
Today : 54ffa04166660fe59f475db04cfccc635fa299f5867312db7982229a6fe2f9f0

Is the reason it breaks because I'm using tag refs instead of a real commit hash ?

lethalman commented:
@roblabla is fetchgit necessary? Use fetchFromGitHub if possible.

roblabla commented:
Fixed, it's using fetchFromGitHub now. Also fixed a failure in pynacl due to missing pytest for the tests to pass. Hopefully, everything should pass now (I didn't catch any problem with daemonize locally).

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -19678,10 +19678,10 @@ in modules // {
       url = "https://pypi.python.org/packages/source/T/Twisted/${name}.tar.bz2";
-      sha256 = "05agfp17cndhv2w0p559lvknl7nv0xqkg10apc47fm53m8llbfvz";
+      sha256 = "0zy18lcrris4aaslil5k12i13k56c32hzfdv6h10kbnzl026h158";
     };
```
Unless you tested with Nox what the effects are of this upgrade I suggest you add a` twisted_15` instead and use that. We can then later see what other packages can use this latest twisted. The same with `simplejson` since that's also a common package. Unless of course you run into collisions.

roblabla commented:
@FRidh I updated twisted separately as suggested. Major bump, so that makes sense.

For simplejson, I went through the changelog and I don't see how any of the changes could break existing applications. They are either new API additions or bugfixes to edge-cases. Is it really okay to create a new package for a minor version bump ?

I also rebased branch on current master, the ids changed again.

FRidh commented:
@roblabla  `nox-review wip` or `nox-review pr 12301` will tell whether a separate package might be needed or not.

roblabla commented:
I get an error running nox-review (both commands) : https://gist.github.com/10f9638a07d367b1f3a0

lethalman commented:
The travis error is related to pypy that is unable to compile, right?

roblabla commented:
Seems like blist doesn't support pypy : https://github.com/DanielStutzbach/blist/issues/68

I added `disabled = isPyPy;` to the package.

lethalman commented:
Built locally, thanks.
